h/accounts/models.py

```diff
@@ -132,6 +133,11 @@ def is_activated(self):
 
         return False
 
+    def activate(self, request):
+        """Activate the user by deleting any activation they have."""
+        request.db.delete(self.activation)
+        request.registry.notify(events.ActivationEvent(request, self))
```

Comment:
I can see the ugliness of doing this in the model class, but I weighed that against duplicating it and chose this one, as it's only this one method that needs to have request passed into it.

The benefit if having a simple `user.activate()` method seems lost to me if you have to call `request.registry.notify(events.ActivationEvent(request, self))` after every time you call `activate()` (and have to add a test for that each time too).

So it seems to come down to wanting model classes to be pure vs wanting to avoid the duplication.

I guess the way to have both things (no duplication, and keep the model class pure) would be to introduce something like an `h.accounts.logic` module to hold an `activate()` function that could be called from multiple different places. But I figured the chances of me getting away with adding that were low :)

I guess th